server/tests/test_notebook_api.py

```diff
@@ -105,3 +108,47 @@ def test_delete_notebook_owner(fake_user, test_notebook, client):
     assert resp.status_code == 204
     assert Notebook.objects.count() == 0
     assert NotebookRevision.objects.count() == 0
+
+
+@pytest.fixture
+def notebook_fork_post_blob():
+    # this blob should be sufficient to create a new notebook (assuming the user of
+    # the api is authorized to do so)
+    return {
+        'title': 'My cool notebook',
+        'content': 'Fake notebook content',
+        'forked_from': 9
+    }
+
+
+@pytest.fixture
+def incorrect_notebook_fork_post_blob():
+    # this blob should be sufficient to create a new notebook (assuming the user of
+    # the api is authorized to do so)
+    return {
+        'title': 'My cool notebook',
+        'content': 'Fake notebook content',
+        'forked_from': 1
+    }
+
+
+def test_fork_notebook_not_logged_in(client, notebook_fork_post_blob):
+    resp = client.post(reverse('notebooks-list'), notebook_fork_post_blob)
+    assert resp.status_code == 403
+
+
+def test_incorrect_fork_notebook(client, fake_user, incorrect_notebook_fork_post_blob):
+    client.force_authenticate(user=fake_user)
+    resp = client.post(reverse('notebooks-list'), incorrect_notebook_fork_post_blob)
+    assert resp.status_code == 400
+
+
+def test_fork_notebook_logged_in(client,
+                                 fake_user,
+                                 fake_user2,
+                                 test_notebook,
+                                 notebook_fork_post_blob):
+    client.force_authenticate(user=fake_user)
+    resp = client.post(reverse('notebooks-list'), notebook_fork_post_blob)
+    assert resp.status_code == 201
+    assert resp.json()['forked_from'] == test_notebook.revisions.get().id
```

Comment:
I would like to see another test that deletes the notebook that another was forked from, verifying that the forked notebook still exists.

Reply:
added this test in a commit below.